forge-game/src/main/java/forge/game/cost/CostPayment.java

```diff
@@ -336,16 +336,18 @@ public static void handleOfferings(final SpellAbility sa, boolean test, boolean
             offering.setUsedToPay(false);
             if (costIsPaid && !test) {
                 sa.getHostCard().getGame().getAction().sacrifice(offering, sa, false, table, null);
```

Comment:
X value will be wrong if toughness was changed by static

- A: still my favorite - don't undo statics on old card when it changes zone (#1922)
- B: check if `changeZoneLKIInfo` can provide it
- C: somehow reuse LKI from ChangesZone trigger
- D: switch value to extra LKI here which feels excessive?

Reply:
I'm going to try to use B, and try to reuse the LKI from ChangeZone

Reply:
i used changeZoneLKIInfo

i tried to use ability.getLastStateBattlefield but it wasn't set at the time
and passing LKI to the function would have been reset later anyway

or your opinion?

Reply:
should be safe 🤷‍♂️